N2-140 Create a Tab mechanism for Advanced gloss page.

jonholdsworth opened the pull request:
Uses Bootstrap library to give Gloss Advanced edit page clickable tabs.
The tabs are currently:
Edit - the familiar edit page with the Enable/Disable Edit button.
Public View - to house the view of the Gloss on the Signbank Dictionary site if it is published there.
Revision History - to house the Revision History of the gloss.
The tabs besides Edit have been left blank internally.
Their respective Jira tickets have been updated with their details to help developers get started.

A video showing the tabs working is attached below:

https://user-images.githubusercontent.com/82071930/176081480-3494957d-08e7-4937-a11f-2ccf1905eda0.mp4

jonholdsworth committed:
N2-140 Create a Tab mechanism for Advanced gloss page.
Uses Bootstrap library to give Gloss Advanced edit page clickable tabs.
The tabs are currently:
Edit - the familiar edit page with the Enable/Disable Edit button.
Public View - to house the view of the Gloss on the Signbank Dictionary site if it is published there.
Revision History - to house the Revision History of the gloss.
The tabs besides Edit have been left blank internally.
Their respective Jira tickets have been updated with their details to help developers get started.

jonholdsworth committed:
N2-140 Create a Tab mechanism for Advanced gloss page.
Uses Bootstrap library to give Gloss Advanced edit page clickable tabs.
The tabs are currently:
Edit - the familiar edit page with the Enable/Disable Edit button.
Public View - to house the view of the Gloss on the Signbank Dictionary site if it is published there.
Revision History - to house the Revision History of the gloss.
The tabs besides Edit have been left blank internally.
Their respective Jira tickets have been updated with their details to help developers get started.

Dimithri committed:
Adding editorial queue (#66)

* Adding editorial queue

* Removing print line

* Changing page heading

* Change wording

jonholdsworth committed:
N2-136 Remove fields from the Advanced gloss page.
Fields removed:
 Top:
  Sign language
 Advanced:
  Phonology
  Relations to Foreign Signs
Code supporting display of these fields has been removed from adminviews.py.

joshmcarthur commented on signbank/dictionary/templates/dictionary/gloss_detail.html:
```diff
@@ -528,4 +541,19 @@ <h4 class="modal-title" id="myModalLabel">{% blocktrans %}Change history for{% e
+
+</div>
+
+{# Tab: Signbank Dictionary public view pane #}
```
It might be a good idea to proactively split each 'tab' out into it's own template partial? This way we can keep the overall size of this template shorter and easy to read

jonholdsworth replied:
I wasn't sure, but now I'll do it. The incoming code is quite small, but you never know it could expand in scope later.

joshmcarthur replied:
I think it's worth doing. This file's already over 500 LOC, so we don't really want this growing more

jonholdsworth replied:
On it.

jonholdsworth committed:
N2-137 Move some fields to bottom of Advanced gloss editing page.
Moved to bottom of page:
  Sign no.
  Filmbatch
  Concise
Moved to just above Relations to Other Signs:
  Gloss Relations

jonholdsworth committed:
N2-137 Move some fields to bottom of Advanced gloss editing page.
Moved to bottom of page:
  Sign no.
  Filmbatch
  Concise
Moved to just above Relations to Other Signs:
  Gloss Relations

jonholdsworth committed:
Typo, left an extra brace behind.

jonholdsworth committed:
Typo, left an extra brace behind.

Dimithri committed:
Adding None to signer (#69)

* Adding None to signer

* Handle missing values better to clear the field state if the value provided is missing or blank

* Refactor signer field to use  a class to add a blank option

* Always sort list items alphabetically

* Remove special treatment of signer field since it now uses the field choice update

Co-authored-by: Josh McArthur <joshua.mcarthur@gmail.com>

Dimithri committed:
Allowing blank for worclasses (#62)

jonholdsworth committed:
Deleted commented-out code.

jonholdsworth committed:
N2-137 Move Advanced gloss editing page fields to bottom.
This requires N2-136 (Remove Advanced gloss editing page fields) to be completed first.
Sign no., Filmbatch and Concise are all moved to the very bottom of the page.
Gloss Relations is now just Relations and is a collapsible field like the others on this page.
It is placed directly above Relations to Other Signs which is more logical location for it.

jonholdsworth committed:
N2-137 Move Advanced gloss editing page fields to bottom.
This requires N2-136 (Remove Advanced gloss editing page fields) to be completed first.
Sign no., Filmbatch and Concise are all moved to the very bottom of the page.
Gloss Relations is now just Relations and is a collapsible field like the others on this page.
It is placed directly above Relations to Other Signs which is more logical location for it.

jonholdsworth committed:
Merge pull request #70 from ODNZSL/task/n2-136-remove-fields-from-advanced

N2-136 Remove fields from the Advanced gloss page.

jonholdsworth committed:
Merge branch 'task/n2-137-move-advanced-fields-to-bottom' of https://github.com/ODNZSL/NZSL-signbank into task/n2-137-move-advanced-fields-to-bottom

jonholdsworth committed:
Merge pull request #71 from ODNZSL/task/n2-137-move-advanced-fields-to-bottom

Task/n2 137 move advanced fields to bottom

jonholdsworth committed:
Splitting Tabs sub-functions into stub files.
The original Advanced gloss page, which sits on the first tab, is left untouched to prevent problems.
The Jira tickets have been updated to reflect the existence of the stub files.

jonholdsworth committed:
Merge branch 'feature/n2-140-tabs-advanced-page' of https://github.com/ODNZSL/NZSL-signbank into feature/n2-140-tabs-advanced-page

jonholdsworth commented:
Closed due to merging mess.
